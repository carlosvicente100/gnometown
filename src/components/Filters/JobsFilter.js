import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { JOBS } from '../../utils/filters'

const JobList = styled.div`
font-family:Helvetica;
display: flex;
flex-wrap: wrap;
background-color:#8395a7;
color:white;
text-align:center;
i{
  color:black;
  margin: 5px 0px;
}
}
`
const JobOption = styled.div`
  margin: 5px 15px;
`

const JobsFilter = ({ list, activeList, onSelect, onClear }) => {
  const handleClick = (event) => {
    onSelect(event.target.value, activeList.includes(event.target.value))
  }
  const handleReset = () => {
    onClear(JOBS)
  }

  const OptionList = () => {
    return list.map((option, key) => {
      return (
        <JobOption key={key} data-testid="job-option">
          <input
            type="checkbox"
            id={option}
            value={option}
            checked={activeList.includes(option)}
            onChange={handleClick}
          ></input>
          <label htmlFor={option}> {option}</label>
        </JobOption>
      )
    })
  }

  return (
    <div>
      {list.length > 0 && (
        <JobList>
          <OptionList></OptionList>
          <i data-testid="job-reset" className="fas fa-eraser" onClick={handleReset}></i>
        </JobList>
      )}
    </div>
  )
}

JobsFilter.propTypes = {
  list: PropTypes.array.isRequired,
  activeList: PropTypes.array.isRequired,
  onSelect: PropTypes.func.isRequired,
  onClear: PropTypes.func.isRequired
}

export default JobsFilter
