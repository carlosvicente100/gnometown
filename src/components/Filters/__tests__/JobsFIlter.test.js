import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import renderer from 'react-test-renderer'

import JobsFilter from '../JobsFilter'

const dummyProps = {
  list: ['Carpenter', 'Farmer', 'Stonecarver', 'Brewer', 'Tax inspector', 'Prospector'],
  activeList: ['Carpenter'],
  onSelect: jest.fn(),
  onClear: jest.fn()
}

describe('JobsFilter Component Suite', () => {
  test('Snapshot check', () => {
    const component = renderer.create(<JobsFilter {...dummyProps} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('JobsFilter Structure', () => {
  test('Should have a checkbox for each list item - prop list, activeList', () => {
    const { getAllByTestId, getByTestId } = render(<JobsFilter {...dummyProps} />)
    const jobList = getAllByTestId('job-option')
    expect(jobList.length).toBe(dummyProps.list.length)
    expect(jobList[0].children[0].checked).toEqual(true)

    const resetButton = getByTestId('job-reset')
    expect(resetButton).toBeTruthy()
  })

  test('Should have a button to reset active filters -  onClear', () => {
    const { getAllByTestId, getByTestId } = render(<JobsFilter {...dummyProps} />)
    const jobList = getAllByTestId('job-option')
    const resetButton = getByTestId('job-reset')

    expect(jobList[0].children[0].checked).toEqual(true)

    fireEvent.click(resetButton)

    expect(dummyProps.onClear).toHaveBeenCalledTimes(1)
  })

  test('Should can active more filters -  onSelect', () => {
    const { getAllByTestId } = render(<JobsFilter {...dummyProps} />)
    const jobList = getAllByTestId('job-option')

    fireEvent.click(jobList[4].children[0])

    expect(dummyProps.onSelect).toHaveBeenCalledTimes(1)
  })
})
