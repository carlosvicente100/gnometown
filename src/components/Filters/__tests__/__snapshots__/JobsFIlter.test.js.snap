// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobsFilter Component Suite Snapshot check 1`] = `
.c0 {
  font-family: Helvetica;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  background-color: #8395a7;
  color: white;
  text-align: center;
}

.c0 i {
  color: black;
  margin: 5px 0px;
}

.c1 {
  margin: 5px 15px;
}

<div>
  <div
    className="c0"
  >
    <div
      className="c1"
      data-testid="job-option"
    >
      <input
        checked={true}
        id="Carpenter"
        onChange={[Function]}
        type="checkbox"
        value="Carpenter"
      />
      <label
        htmlFor="Carpenter"
      >
         
        Carpenter
      </label>
    </div>
    <div
      className="c1"
      data-testid="job-option"
    >
      <input
        checked={false}
        id="Farmer"
        onChange={[Function]}
        type="checkbox"
        value="Farmer"
      />
      <label
        htmlFor="Farmer"
      >
         
        Farmer
      </label>
    </div>
    <div
      className="c1"
      data-testid="job-option"
    >
      <input
        checked={false}
        id="Stonecarver"
        onChange={[Function]}
        type="checkbox"
        value="Stonecarver"
      />
      <label
        htmlFor="Stonecarver"
      >
         
        Stonecarver
      </label>
    </div>
    <div
      className="c1"
      data-testid="job-option"
    >
      <input
        checked={false}
        id="Brewer"
        onChange={[Function]}
        type="checkbox"
        value="Brewer"
      />
      <label
        htmlFor="Brewer"
      >
         
        Brewer
      </label>
    </div>
    <div
      className="c1"
      data-testid="job-option"
    >
      <input
        checked={false}
        id="Tax inspector"
        onChange={[Function]}
        type="checkbox"
        value="Tax inspector"
      />
      <label
        htmlFor="Tax inspector"
      >
         
        Tax inspector
      </label>
    </div>
    <div
      className="c1"
      data-testid="job-option"
    >
      <input
        checked={false}
        id="Prospector"
        onChange={[Function]}
        type="checkbox"
        value="Prospector"
      />
      <label
        htmlFor="Prospector"
      >
         
        Prospector
      </label>
    </div>
    <i
      className="fas fa-eraser"
      data-testid="job-reset"
      onClick={[Function]}
    />
  </div>
</div>
`;
